firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -0,0 +1,246 @@
+  public InstallationTokenResult generateAuthToken(
+      long projectNumber, @NonNull String apiKey, @NonNull String fid, @NonNull String refreshToken)
+      throws FirebaseException {
+    String resourceName =
+        String.format(GENERATE_AUTH_TOKEN_REQUEST_RESOURCE_NAME_FORMAT, projectNumber, fid);
+    try {
+      URL url =
+          new URL(
+              String.format(
+                  "https://%s/%s/%s?key=%s",
+                  FIREBASE_INSTALLATIONS_API_DOMAIN,
+                  FIREBASE_INSTALLATIONS_API_VERSION,
+                  resourceName,
+                  apiKey));
+
+      HttpsURLConnection httpsURLConnection = (HttpsURLConnection) url.openConnection();
+      httpsURLConnection.setDoOutput(true);
+      httpsURLConnection.setRequestMethod("POST");
+      httpsURLConnection.addRequestProperty("Authorization", "FIS_V2 " + refreshToken);
+      httpsURLConnection.addRequestProperty(CONTENT_TYPE_HEADER_KEY, JSON_CONTENT_TYPE);
+      httpsURLConnection.addRequestProperty(CONTENT_ENCODING_HEADER_KEY, GZIP_CONTENT_ENCODING);
+
+      int httpResponseCode = httpsURLConnection.getResponseCode();
+      switch (httpResponseCode) {
+        case 200:
+          return readGenerateAuthTokenResponse(httpsURLConnection);
+        case 401:
+          throw new FirebaseException("The request did not have the required credentials.");
+        default:
+          throw new FirebaseException("There was an internal server error.");
+      }
+    } catch (IOException e) {
+      throw new FirebaseException("The server returned an unexpected error.: " + e.getMessage());
+    }
+  }
+  // Read the response from the createFirebaseInstallation API.
+  private InstallationResponse readCreateResponse(HttpsURLConnection conn) throws IOException {
+    JsonReader reader =
+        new JsonReader(new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()));
+    InstallationTokenResult.Builder installationTokenResult = InstallationTokenResult.builder();
+    InstallationResponse.Builder builder = InstallationResponse.builder();
+    reader.beginObject();
+    while (reader.hasNext()) {
+      String name = reader.nextName();
+      if (name.equals("name")) {
+        builder.setName(reader.nextString());
+      } else if (name.equals("refreshToken")) {
+        builder.setRefreshToken(reader.nextString());
+      } else if (name.equals("authToken")) {
+        reader.beginObject();
+        while (reader.hasNext()) {
+          String key = reader.nextName();
+          if (key.equals("token")) {
+            installationTokenResult.setAuthToken(reader.nextString());
+          } else if (key.equals("expiresIn")) {
+            installationTokenResult.setTokenExpirationTimestampMillis(reader.nextLong());
+          } else {
+            reader.skipValue();
+          }
+        }
+        builder.setAuthToken(installationTokenResult.build());
+        reader.endObject();
+      } else {
+        reader.skipValue();
+      }
```

Comment:
If you want http to respond in json format, you need to add this header to your request.
https://stackoverflow.com/questions/43209924/rest-api-use-the-accept-application-json-http-header

Reply:
Done. Thanks.